src/qibosoq/programs/flux.py

```diff
@@ -84,3 +86,16 @@ def body(self):
         self.set_bias("zero")
         self.soc.reset_gens()
         self.sync_all(self.relax_delay)
+
+    def declare_zones_and_ro(self, sequence: List[Pulse]):
+        """Declare all nqz zones and readout frequencies.
+
+        Declares drives, fluxes and readout (mux or not) and readout freq.
+        """
+        self.declare_nqz_zones([pulse for pulse in sequence if pulse.type == "drive"])
+        self.declare_nqz_flux()
```

Comment:
And how does it work this empty declaration?
(and the following as well)

Since it's taking no arguments and returns nothing, this should modify somehow `self`, but I wonder what is doing...

Reply:
It considers the self.qubits attribute and sees which qubits are biased.
In #29 , this declaration is also extended to flux pulses

Reply:
What do you mean?
 Both the qubits and the program exist only in qibosoq... I can move the function call somewhere else, but in any case I will have to write it